sdk/digitaltwins/digital-twins-core/test/public/testDigitalTwins.spec.ts

```diff
@@ -0,0 +1,939 @@
+    const patch = [
+      {
+        op: "replace",
+        path: "/AverageTemperature",
+        value: 42
+      }
+    ];
+    await client.upsertDigitalTwin(digitalTwinId, JSON.stringify(buildingTwin));
+
+    let options: DigitalTwinsDeleteOptionalParams = {
+      ifMatch: "XXX"
+    };
+    let errorWasThrown = false;
+    try {
+      await client.updateDigitalTwin(digitalTwinId, patch, options);
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(
+        error.message,
+        `Invalid If-Match header value. Allowed value(s): If-Match: {etag} or If-Match: *`
+      );
+    } finally {
+      await deleteDigitalTwin(digitalTwinId);
+      await deleteModels();
+    }
+    should.equal(errorWasThrown, true, "Error was not thrown");
+  });
+
+  it("update digital twin not existing", async function() {
+    const digitalTwinId = recorder.getUniqueName("digitalTwin", "update-digitaltwin-not-existing");
+
+    await setUpModels();
+    await deleteDigitalTwin(digitalTwinId);
+
+    const patch = [
+      {
+        op: "replace",
+        path: "/AverageTemperature",
+        value: 42
+      }
+    ];
+
+    let errorWasThrown = false;
+    try {
+      await client.updateDigitalTwin(digitalTwinId, patch);
+    } catch (error) {
+      errorWasThrown = true;
+      assert.include(error.message, `There is no digital twin instance that exists with the ID`);
+    } finally {
+      await deleteDigitalTwin(digitalTwinId);
+      await deleteModels();
+    }
+    should.equal(errorWasThrown, true, "Error was not thrown");
+  });
+
+  it("query digital twin", async function() {
+    const digitalTwinId = recorder.getUniqueName("digitalTwin", "query-digitaltwin");
+
+    await setUpModels();
+    await deleteDigitalTwins();
+
+    const buildingTwin = {
+      $metadata: {
+        $model: BUILDING_MODEL_ID
+      },
+      AverageTemperature: 68,
+      TemperatureUnit: "Celsius"
+    };
+    await client.upsertDigitalTwin(digitalTwinId, JSON.stringify(buildingTwin));
+
+    // Wait for the service to be ready
+    await delay(5000);
+
+    let twinFound = false;
+    try {
+      const query = "SELECT * FROM digitaltwins";
+      const queryResult = client.queryTwins(query);
+      for await (const item of queryResult) {
+        if (item.$dtId == digitalTwinId) {
+          twinFound = true;
+          break;
+        }
+      }
+      assert.equal(twinFound, true, "DigitalTwin did not find");
+    } finally {
+      await deleteDigitalTwin(digitalTwinId);
+      await deleteModels();
+    }
+  });
+
+  it("query digital twin invalid expression", async function() {
+    const digitalTwinId = recorder.getUniqueName(
+      "digitalTwin",
+      "query-digitaltwin-invalid-expression"
+    );
+
+    await setUpModels();
+    await deleteDigitalTwins();
+
+    let errorWasThrown = false;
+    try {
+      const query = "foo";
+      const queryResult = client.queryTwins(query);
+      for await (const _ of queryResult) {
```

Comment:
This fails in node 8 because of the usage of async iterators since they are an experimental feature in node 8.

Can you try appending the `unit-test:node` and `integration-test:node` commands with `--harmony` flag to get rid of the build failures?

Reply:
Done